lib/SGN/Controller/AJAX/BreedersToolbox/Population.pm

```diff
@@ -25,10 +25,32 @@ sub create_population :Path('/ajax/population/new') Args(0) {
     my $self = shift;
     my $c = shift;
     my $schema = $c->dbic_schema("Bio::Chado::Schema");
+    my $session_id = $c->req->param("sgn_session_id");
```

Comment:
since you have 3 places where you check this I suggest to move the entire block to a separate function